### ISSUE TITLE
ATLAS Open Data  𝐻→𝑍𝑍 Analysis Grand Challenge notebook requires scipy v1.6.0+

### ISSUE THREAD

**Issue by matthewfeickert** [CoffeaTeam/coffea-casa#379]
In `v1.0.0` of the [Analysis Grand Challenge ATLAS Open Data 𝐻→𝑍𝑍 notebook](https://github.com/iris-hep/analysis-grand-challenge/blob/cdb9e674544b2ab8b12ad765c59c01510f4ac301/analyses/atlas-open-data-hzz/HZZ_analysis_pipeline.ipynb) the 

```python
mu_test = 1.0
cls = float(pyhf.infer.hypotest(mu_test, data, model))
print(f"CL_S for Signal_norm={mu_test} is {cls:.3f}")
```

cell will hang as there is an optimization issue in `scipy` `v1.5.x`, and the coffea-casa release at UChicago (as of 2023-05-03)

![uchicago-af-coffea-casa-launch](https://user-images.githubusercontent.com/5142394/236042411-f94fa55a-6863-4380-be4b-4e90592667ec.png)

has `scipy` `v1.5.3`. This distribution has `pyhf` `v0.7.1` and so has a lower bound on scipy of `v1.5.1` but there must be some other requirement from

```console
$ python -m pip show scipy
Name: scipy
Version: 1.5.3
Summary: SciPy: Scientific Library for Python
Home-page: https://www.scipy.org/
Author: 
Author-email: 
License: BSD
Location: /opt/conda/lib/python3.8/site-packages
Requires: numpy
Required-by: cabinetry, coffea, dask-glm, dask-ml, mlflow, pyhf, scikit-learn, shap, torch-cluster, torch-geometric, torch-sparse, xgboost
```

that is limiting the `scipy` release to `v1.5.3`.

**Comment by matthewfeickert** [comment]
> cell will hang as there is an optimization issue in `scipy` `v1.5.x`

That is apparently not correct, as both @alexander-held and I have been able to find that if a user does

```
python -m pip uninstall --yes scipy
python -m pip install --upgrade 'scipy<=1.5.3'
```

and then reruns the notebook things pass. :( Very confusing, but I can only assume that there is some problem with the conda distribution of `scipy` `v1.5.3`?

**Comment by alexander-held** [comment]
Smaller reproducer: https://gist.github.com/alexander-held/75ef9cb15c14e0ad610950197bccdac1 (haven't tried other workspaces yet, wouldn't be surprised if it is not model-specific).

**Comment by matthewfeickert** [comment]
Thanks @alexander-held. I think this really is something to do with the mamba distribution as if you reinstall `scipy` `v1.5.3` from PyPI things are fixed but if you `mamba update 'scipy==1.5.3'` then things fail.

**Comment by matthewfeickert** [comment]
Working from fitting versions back on the UChicago

```console
$ git clone https://gist.github.com/75ef9cb15c14e0ad610950197bccdac1.git alex-reproducer
$ cd alex-reproducer
```

then the limit seems to be at `v1.8.1` as

```console
$ mamba update --channel conda-forge --yes 'scipy==1.8.0'
$ python fit.py
```

fails, and 

```console
$ mamba update --channel conda-forge --yes 'scipy==1.8.1'
$ python fit.py
```
passes.
